src/server/components/lists/controllers/ingest/__tests__/helpers.test.ts

```diff
@@ -1,17 +1,27 @@
 import { arrayHasChanges, getObjectDiff } from "../helpers";
 
-test("arrayHasChanges", () => {
-  expect(arrayHasChanges([1], [1])).toBeFalsy();
-  expect(arrayHasChanges(["a"], ["a"])).toBeFalsy();
-  expect(arrayHasChanges([], [])).toBeFalsy();
-  expect(arrayHasChanges(["fried", "scrambled", "poached"], ["fried", "scrambled", "poached"])).toBeFalsy();
+describe("arrayHasChanges", () => {
+  it("returns false for arrays with the same value in the same order", () => {
+    expect(arrayHasChanges([1], [1])).toBeFalsy();
+    expect(arrayHasChanges(["a"], ["a"])).toBeFalsy();
+    expect(arrayHasChanges([], [])).toBeFalsy();
+    expect(arrayHasChanges(["fried", "scrambled", "poached"], ["fried", "scrambled", "poached"])).toBeFalsy();
+    expect(arrayHasChanges([true, true, false], [true, true, false])).toBeFalsy();
+  });
 
-  expect(arrayHasChanges(["a", "b", "c"], ["b", "c", "a"])).toBeFalsy();
-  expect(arrayHasChanges([1, 2, 3], [3, 1, 2])).toBeFalsy();
+  it("returns false for arrays with the same value but different order", () => {
+    expect(arrayHasChanges(["a", "b", "c"], ["b", "c", "a"])).toBeFalsy();
+    expect(arrayHasChanges([1, 2, 3], [3, 1, 2])).toBeFalsy();
+    expect(arrayHasChanges([true, true, false], [true, false, true])).toBeFalsy();
+  });
 
-  expect(arrayHasChanges([1], [1, 2])).toBeTruthy();
-  expect(arrayHasChanges([1, 2], [1])).toBeTruthy();
-  expect(arrayHasChanges(["a", "b", "c"], ["d", "e", "f"])).toBeTruthy();
+  it("returns true for arrays with different values", () => {
```

Comment:
Looking at arrayHasChanges I think it will return falsy if beforeArray = [] and afterArray is populated with something.

https://github.com/UKForeignOffice/lists/blob/fd5f28b9c6653784918edfcd22181bb22f2eb8c1/src/server/components/lists/controllers/ingest/helpers.ts#L17